src/lib/mixer_module/mixer_module.cpp

```diff
@@ -270,54 +349,76 @@ bool MixingOutput::update()
 
 	/* do mixing */
 	float outputs[MAX_ACTUATORS] {};
-	const unsigned mixed_num_outputs = _mixers->mix(outputs, MAX_ACTUATORS);
+	const unsigned mixed_num_outputs = _mixers->mix(outputs, _max_num_outputs);
 
 	/* the output limit call takes care of out of band errors, NaN and constrains */
-	uint16_t output_limited[MAX_ACTUATORS] {};
-
 	output_limit_calc(_throttle_armed, armNoThrottle(), mixed_num_outputs, _reverse_output_mask,
-			  _disarmed_value, _min_value, _max_value, outputs, output_limited, &_output_limit);
+			  _disarmed_value, _min_value, _max_value, outputs, _current_output_value, &_output_limit);
 
 	/* overwrite outputs in case of force_failsafe with _failsafe_value values */
 	if (_armed.force_failsafe) {
 		for (size_t i = 0; i < mixed_num_outputs; i++) {
-			output_limited[i] = _failsafe_value[i];
+			_current_output_value[i] = _failsafe_value[i];
 		}
 	}
 
+	bool stop_motors = mixed_num_outputs == 0 || !_throttle_armed;
+
 	/* overwrite outputs in case of lockdown or parachute triggering with disarmed values */
 	if (_armed.lockdown || _armed.manual_lockdown) {
 		for (size_t i = 0; i < mixed_num_outputs; i++) {
-			output_limited[i] = _disarmed_value[i];
+			_current_output_value[i] = _disarmed_value[i];
 		}
-	}
-
-	bool stop_motors = true;
 
-	if (mixed_num_outputs > 0) {
-		/* assume if one (here the 1.) motor is disarmed, all of them should be stopped */
-		stop_motors = (output_limited[0] == _disarmed_value[0]);
+		stop_motors = true;
 	}
 
 	/* apply _param_mot_ordering */
-	reorderOutputs(output_limited);
+	reorderOutputs(_current_output_value);
 
 	/* now return the outputs to the driver */
-	_interface.updateOutputs(stop_motors, output_limited, mixed_num_outputs, n_updates);
+	if (_interface.updateOutputs(stop_motors, _current_output_value, mixed_num_outputs, n_updates)) {
+		actuator_outputs_s actuator_outputs{};
+		setAndPublishActuatorOutputs(mixed_num_outputs, actuator_outputs);
 
+		publishMixerStatus(actuator_outputs);
+		updateLatencyPerfCounter(actuator_outputs);
+	}
+
+	checkSafetyButton();
+	handleCommands();
+
+	return true;
+}
 
-	actuator_outputs_s actuator_outputs{};
-	actuator_outputs.noutputs = mixed_num_outputs;
+void
+MixingOutput::checkSafetyButton()
+{
+	if (_safety_sub.updated()) {
+		safety_s safety;
+
+		if (_safety_sub.copy(&safety)) {
```

Comment:
I though you can do:
```
saftey_s safety;
if (_safety_sub.update(safety)) {
    _safety_off = !safety.safety_switch_available || safety.safety_off;
}
```

Reply:
I'm not sure if safety is needed here at all. We really should be able to trust the state machine and `actuator_armed` which factors in `safety` (and the kill switch, etc). The safety message isn't even logged.

Reply:
Indeed, I removed it in https://github.com/PX4/Firmware/pull/13205/commits/a1d55036a6bf2bdfde73ebf37c56a98f70d76865